settings/src/components/password.js

```diff
@@ -27,11 +27,7 @@ import { GlobalContext } from '../script';
 export default function Password() {
 	const { setGlobalNotice, userRecord } = useContext( GlobalContext );
 	const [ inputType, setInputType ]     = useState( 'password' );
-	let passwordStrong                    = true; // Saved passwords have already passed the test.
-
-	if ( userRecord.hasEdits ) {
-		passwordStrong = isPasswordStrong( userRecord.editedRecord.password, userRecord.record );
-	}
+	const [ passwordStrong, setPasswordStrong ] = useState( true );
```

Comment:
I'm not sure that this and the related changes should be necessary, see 0e9017eef0a7c5d24156e11d7c1771fcdcf8849d. Curios to hear your thoughts though 'cause I could easily be wrong.

Reply:
Yeah, that's true, as it sits, we don't technically need to add a state variable for password strength since the provider is going re-render. I ended up making these changes while looking into #53. 

Since that API could  kick back a weak password error, we'll want to reshow the weak password notice. To do so, I'm thinking we'll need to wrap the `await userRecord.save()` in a `try/catch` and retrigger the weak password notice. The way this is currently working, we won't be able to do so since it only react to changes in the provider.  

**Example**
```
const savePasswordHandler = useCallback(async () => {
  try {
    await userRecord.save();

    const response = await apiFetch({
      url: apiFetch.nonceEndpoint,
      parse: false,
    });
    apiFetch.nonceMiddleware.nonce = await response.text();

    setGlobalNotice("New password saved.");
  } catch (e) {
    // check for weak password server error
    if (e.code === "password_too_weak") {
        setPasswordStrong( false )
    }

  }
}, []);
```
Does that make sense? Is there a different approach that you think may work?

Reply:
Ah, that makes sense 👍🏻 Ideally those changes would be in a separate PR, but it's not a big deal if it's easier to do here. It'd be helpful for the final commit message to document that, though.

When we catch the error, will be need to create a state for the error message, in order to display it to the user? If so, then we may not need one for `passwordStrong`, since we could do something like:

```js
if ( userRecord.hasEdits ) {
	passwordStrong = isPasswordStrong( userRecord.editedRecord.password, userRecord.record );
} elseif( 'password_too_weak' === passwordSaveError.code ) {
	passwordString = false;
}
```

Reply:
Yep, fair enough. I've pulled the changes from this PR. We can move the discussion over to the next one. 👍